app/lib/webfinger_resource.rb

```diff
@@ -0,0 +1,60 @@
+class WebfingerResource
+  attr_reader :resource
+
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def username
+    case resource
+    when /https/
```

Comment:
Although your tests seem to account for the http case so I'm not sure what this regex does